Mourning Chorus resolution pruning

Addisonstumpf opened the pull request:
Prune Mourning Chorus resolution options to only display defended locations (for rallying), and to only display the elites corresponding to infected locations (for elimination).

zzorba commented:
I'm not totally sure, but I suspect we need to version this change.

I think despite the fact I have IDs, I'm storing choices based on indexes (bad, but hard to change at this point). So removing indexes conditionally might break this for existing campaigns.

If youve found otherwise I'd believe it, but the nagging voice in the back of my head says versioning this is safer.

Addisonstumpf commented:
"safe than sorry" seems like a good rule for that kind of thing, sure. Nowhere near "having a personal understanding of versioning", let alone "opposing your understanding of versioning", ha.

Addisonstumpf commented:
similar stuff needs to happen in (at least) Afternoon War, should probably lump those together instead of repeatedly running up the versioning. Will undraft when ready.

zzorba commented on campaigns/zcp/mourning_chorus.json:
```diff
@@ -2005,10 +2005,277 @@
+          {
+            "boolCondition": false,
+            "steps": [
+              "record_rallied_legacy"
```
So the old one should keep the name "record_rallied" while the new one will need the new name ("record_rallied_new") -- basically we want to maintain the ID for the old version on the input.

zzorba commented on campaigns/zcp/mourning_chorus.json:
```diff
@@ -2168,10 +2435,233 @@
+          {
+            "boolCondition": true,
+            "steps": [
+              "record_eliminated_elites"
```
Same here, old version should keep the original name, new one needs a new name.

zzorba commented on campaigns/zcp/the_afternoon_war.json:
```diff
@@ -1158,8 +1158,299 @@
+          {
+            "boolCondition": false,
+            "steps": [
+              "record_eliminated_elites_legacy"
```
We should flip these names, same as above.

Addisonstumpf commented:
Checked, I had changed old names in the previous round of Mourning Chorus changes. Updated to A, revert that, and B, rename the new parts to use suffixes instead of prefixes. This feels like it probably also sets off a versioning problem, open to suggestions for how to minimize the wound here.
